### PR TITLE
Combined PR: NEMS Driver cleanup (#533), Wave update (#542), add CPLD&DATM tests on Jet (#545)

### PR DESCRIPTION
**Pull request by DeniseWorthen**
# PR Checklist

- [x] Ths PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [X] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [ ] If new or updated input data is required by this PR, it is clearly stated in the text of the PR.

## Instructions: All subsequent sections of text should be filled in as appropriate.

The information provided below allows the code managers to understand the changes relevant to this PR, whether those changes are in the ufs-weather-model repository or in a subcomponent repository. Ufs-weather-model code managers will use the information provided to add any applicable labels, assign reviewers and place it in the Commit Queue. Once the PR is in the Commit Queue, it is the PR owner's responsiblity to keep the PR up-to-date with the develop branch of ufs-weather-model. 

## Description

- Removes NEMS mediator and unused or legacy components as model options
- Implements fd_nems.yaml by default (replaces FieldDictionaryAdd)
- Cleans up error checking in driver

### Issue(s) addressed

NEMS issue [#95](https://github.com/NOAA-EMC/NEMS/issues/95)
Includes PR #542 
Includes PR #545 

## Testing

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

Pre-commit testing shows these changes are B4B for both gnu and intel on Hera. 

Also tested on Gaea after addition of Wave and Jet PRs and all tests passed.

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3

## Dependencies

NEMS PR [#98](https://github.com/NOAA-EMC/NEMS/pull/98)

co-authors: 
@JessicaMeixner-NOAA 
@binli2337

### CODE REVIEW

**Review by DusanJovic-NOAA — APPROVED** [review]

**Comment by DeniseWorthen** [comment]
@binli2337 I've merged the latest ufs-weather so we're ready for you to create the jet baselines. Please double check the rt.conf. Since the last PR changed the order I had to be careful about merging. I think I got it right.

**Comment by binli2337** [comment]
@DeniseWorthen New baselines for coupled and datm tests have been created on Jet.

**Comment by DeniseWorthen** [comment]
Thanks Bin! I'll go ahead an kick off the RTs for this PR.

**Comment by BrianCurtis-NOAA** [comment]
Machine: hera
Compiler: gnu
Job: RT
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/618081083/20210501200042/ufs-weather-model
Please make changes and add the following label back:
hera-gnu-RT

**Comment by BrianCurtis-NOAA** [comment]
Machine: hera
Compiler: intel
Job: RT
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/618081083/20210501201004/ufs-weather-model
Please manually delete: /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_308327
Test cpld_control_wave 027 failed in check_result failed 
Test cpld_control_wave 027 failed in run_test failed 
Test cpld_bmarkfrac_wave_v16 026 failed in check_result failed 
Test cpld_bmarkfrac_wave_v16 026 failed in run_test failed 
Test cpld_bmarkfrac_wave 025 failed in check_result failed 
Test cpld_bmarkfrac_wave 025 failed in run_test failed 
Test cpld_bmark_wave 024 failed in check_result failed 
Test cpld_bmark_wave 024 failed in run_test failed 
Test fv3_wrtGauss_nemsio_c768 110 failed in run_test failed 
Test fv3_gfdlmprad 108 failed in check_result failed 
Test fv3_gfdlmprad 108 failed in run_test failed 
Test fv3_gfdlmprad_atmwav 109 failed in check_result failed 
Test fv3_gfdlmprad_atmwav 109 failed in run_test failed 
Please make changes and add the following label back:
hera-intel-RT

**Comment by DeniseWorthen** [comment]
@binli2337 The hera intel and gnu passed so if you haven't started the replacement baselines on jet yet I think everything should now be correct and you can start.

@DeniseWorthen Three tests have completed.

File: tests/default_vars.sh [hunk 1]
```diff
@@ -161,6 +161,42 @@ elif [[ $MACHINE_ID = jet.* ]]; then
...
 
+  TASKS_cpl_dflt=192; TPN_cpl_dflt=24; INPES_cpl_dflt=3; JNPES_cpl_dflt=8
+  THRD_cpl_dflt=1; WPG_cpl_dflt=6;  MPB_cpl_dflt="0 143"; APB_cpl_dflt="0 149"
+  OPB_cpl_dflt="150 179"; IPB_cpl_dflt="180 191"
+
+  TASKS_cpl_dflt_wwav=204; TPN_cpl_dflt_wwav=24; INPES_cpl_dflt_wwav=3; JNPES_cpl_dflt_wwav=8
+  THRD_cpl_dflt_wwav=1; WPG_cpl_dflt_wwav=6;  MPB_cpl_dflt_wwav="0 143"; APB_cpl_dflt_wwav="0 149"
+  OPB_cpl_dflt_wwav="150 179"; IPB_cpl_dflt_wwav="180 191"; WPB_cpl_dflt_wwav="192 203"
+
+  TASKS_cpl_thrd=120; TPN_cpl_thrd=18; INPES_cpl_thrd=3; JNPES_cpl_thrd=4
```
Comment (MinsukJi-NOAA) [inline tests/default_vars.sh:172]: How is TPN_cpl_thrd=18 related to TPN_cpl_dflt=24? On a related note, I see that TPN_cpl_dflt=40 and TPN_cpl_thrd=40 for, example, Hera. Is this intended (cf. TPN_dflt=40 and TPN_thrd=20 for Hera)?

Comment (DeniseWorthen) [reply tests/default_vars.sh:172]: I had noticed that also in Bin's original PR. They're not consistent across platforms. I had created an issue #320 a while ago as a way of getting around this problem of having to manually edit the numbers for the coupled model.

**Comment by MinsukJi-NOAA** [comment]
CI test passed for 4f593f9: https://github.com/ufs-community/ufs-weather-model/actions/runs/807177711

**Review by MinsukJi-NOAA — APPROVED** [review]

**Review by junwang-noaa — APPROVED** [review]

**Comment by DeniseWorthen** [comment]
@binli2337 Are we ready to run the auto-RT on Jet?

**Comment by binli2337** [comment]
@DeniseWorthen New baselines are ready on Jet.

**Comment by DeniseWorthen** [comment]
All RTs have completed and all passed. This is ready to merge after NEMS is updated and I revert the submodule.

**Pull request merged**